cbcontainers/state/state_applier.go

```diff
@@ -256,31 +256,40 @@ func (c *StateApplier) applyEnforcerWebhooks(ctx context.Context, agentSpec *cbc
 		return false, err
 	}
 
-	c.enforcerMutatingWebhook.UpdateTlsSecretValues(tlsSecretValues)
-	mutatedMutatingWebhook, _, err := c.applier.Apply(ctx, c.enforcerMutatingWebhook, agentSpec, applyOptions)
-	if err != nil {
-		return false, err
+	mutatedMutatingWebhook := false
+	if agentSpec.Components.Basic.Enforcer.EnableEnforcingWebhook {
+		c.enforcerMutatingWebhook.UpdateTlsSecretValues(tlsSecretValues)
+		mutatedMutatingWebhook, _, err = c.applier.Apply(ctx, c.enforcerMutatingWebhook, agentSpec, applyOptions)
+		if err != nil {
+			return false, err
+		}
+	} else {
+		// Note that if someone changes the flag in sequence false -> true -> false, we should delete the webhook again
+		deletedWebhookDueToDisabledFlag, deleteErr := c.deleteSingleEnforcerWebhook(ctx, agentSpec, c.enforcerMutatingWebhook)
```

Comment:
This also runs when the flag was always false, but the deletion is a no-op if the object does not exist so I think it is OK

Reply:
agree